state/protocol/snapshot.go

```diff
@@ -102,11 +104,9 @@ type Snapshot interface {
 
 	// RandomSource returns the source of randomness derived from the Head block.
```

Comment:
I think this is a very subtle topic that we need to carefully explain.
```suggestion
	// RandomSource returns the source of randomness _for_ the snapshot's Head block. 
	// Note that the source of randomness for a block `H`, is contained in the 
	// QuorumCertificate [QC] for block `H` (QCs for H are distributed as part of child
	// blocks, timeout messages or timeout certificates). While there might be different 
	// QCs for block H, they all yield exactly the same source of randomness (feature of 
	// threshold signatures used here). Therefore, it is a possibility that there is no 
	// QC known (yet) for the head block. 
```